File: backend/internal/models/invoice.go
package models

import (
	"encoding/json"
	"time"
)

// Invoice represents a cloud invoice synced from MOF API
type Invoice struct {
	ID                      uint             `gorm:"primaryKey" json:"id"`
	UserID                  uint             `gorm:"not null;index" json:"user_id"`
	InvoiceNumber           string           `gorm:"not null;size:10" json:"invoice_number"`
	InvoiceDate             time.Time        `gorm:"not null" json:"invoice_date"`
	SellerName              string           `gorm:"size:255" json:"seller_name"`
	SellerBAN               string           `gorm:"size:8;column:seller_ban" json:"seller_ban"`
	Amount                  float64          `gorm:"type:decimal(10,2);not null" json:"amount"`
	Status                  string           `gorm:"size:50" json:"status"`
	Items                   json.RawMessage  `gorm:"type:jsonb" json:"items"`
	IsDuplicated            bool             `gorm:"default:false" json:"is_duplicated"`
	DuplicatedTransactionID *uint            `json:"duplicated_transaction_id,omitempty"`
	ConfidenceScore         *float64         `gorm:"type:decimal(3,2)" json:"confidence_score,omitempty"`
	CreatedAt               time.Time        `json:"created_at"`

	User                    User             `gorm:"foreignKey:UserID" json:"-"`
	DuplicatedTransaction   *Transaction     `gorm:"foreignKey:DuplicatedTransactionID" json:"duplicated_transaction,omitempty"`
}

func (Invoice) TableName() string {
	return "invoices"
}

// InvoiceItem represents an item in an invoice
type InvoiceItem struct {
	Description string  `json:"description"`
	Quantity    string  `json:"quantity"`
	UnitPrice   string  `json:"unit_price"`
	Amount      string  `json:"amount"`
}

// InvoiceSyncRequest represents the request to sync invoices
type InvoiceSyncRequest struct {
	StartDate string `json:"start_date" binding:"required"` // YYYY/MM/DD
	EndDate   string `json:"end_date" binding:"required"`   // YYYY/MM/DD
}

// InvoiceSettingsInput represents input for updating invoice settings
type InvoiceSettingsInput struct {
	InvoiceCarrier string `json:"invoice_carrier" binding:"required"`
}

// ConfirmDuplicateRequest represents input for confirming a duplicate
type ConfirmDuplicateRequest struct {
	InvoiceID     uint `json:"invoice_id" binding:"required"`
	TransactionID uint `json:"transaction_id" binding:"required"`
}

File: frontend/src/types/invoice.ts
export interface Invoice {
  id: number
  user_id: number
  invoice_number: string
  invoice_date: string
  seller_name: string
  seller_ban: string
  amount: number
  status: string
  items: InvoiceItem[] | null
  is_duplicated: boolean
  duplicated_transaction_id?: number
  confidence_score?: number
  created_at: string
  duplicated_transaction?: {
    id: number
    amount: number
    description: string
    transaction_date: string
    source: string
  }
}

export interface InvoiceItem {
  description: string
  quantity: string
  unit_price: string
  amount: string
}

export interface InvoiceFilter {
  start_date?: string
  end_date?: string
  page?: number
  page_size?: number
}

export interface InvoiceListResponse {
  invoices: Invoice[]
  total: number
  page: number
  page_size: number
}

export interface InvoiceSyncRequest {
  start_date: string
  end_date: string
}

export interface InvoiceSyncResponse {
  synced: number
  message: string
}

export interface ConfirmDuplicateRequest {
  invoice_id: number
  transaction_id: number
}

export interface InvoiceSettingsInput {
  invoice_carrier: string
}
